### PR TITLE
Fix/path smoother

### PR DESCRIPTION
**Pull request by sim0504**

### CODE REVIEW

File: RULEngine/filters/path_smoother.py [hunk 1]
```diff
@@ -1,104 +1,97 @@
...
+
+def filter_points_and_speed(point_list, speed_list, threshold):
     position_list = []
     new_speed_list = []
-    new_turns_list = []
-    for idx, point in enumerate(point_list[0:-1]):
-        i = idx
-        if (point_list[i] - point_list[i+1]).norm < 100:
+
+    for i, point in enumerate(point_list[0:-1]):
```
[the remark this answers is not in the document]
Comment (sim0504) [reply RULEngine/filters/path_smoother.py]: can't , I need the indice to filter the speeds list...

**Comment by sim0504** [comment]
I've add many things. I should have done another branch sorry! Pathfinder Module is now a thing.

File: RULEngine/Debug/uidebug_command_factory.py [hunk 2]
```diff
@@ -121,22 +121,43 @@ def robot_state(state):
...
+                                                       timeout=0.1))
+
+            # MultiplePoints is weird, it has a special behavior were an unique ID link must be provided
+            cmds.append(UIDebugCommandFactory.multiple_points(path.points[1:],
+                                                              BLUE,
+                                                              width=5,
+                                                              link="raw_path - " + str(robot.robot_id),
+                                                              timeout=0.0))
+
+        for robot in robots:
```
Comment (sim0504) [inline RULEngine/Debug/uidebug_command_factory.py]: sorry was lazy.

File: ai/executors/pathfinder_module.py [hunk 3]
```diff
@@ -1,42 +1,52 @@
...
-
-    return PathPartitionner()
-
-
-def generate_path(game_state, player, ai_command):
-
-    pathfinder = create_pathfinder()
-
-    collision_bodies = get_pertinent_collision_objects(player, game_state, ai_command)
+class PathfinderModule(metaclass=Singleton):
```
Comment (PhiBabin) [inline ai/executors/pathfinder_module.py]: Since only PlayExecutor use PathfinderModule, I don't think it need to be a singleton.

Comment (sim0504) [reply ai/executors/pathfinder_module.py]: oups my bad, copy-paste

File: Util/engine_command.py [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
+                           'path,'
+                           'kick_type,'
+                           'kick_force,'
+                           'dribbler_active,'
+                           'charge_kick,'
+                           'target_orientation,'
+                           'end_speed,')
+
+
+def generate_engine_cmd(player: Player, ai_cmd: AICommand, path):
```
Comment (PhiBabin) [inline Util/engine_command.py]: Je pense que ce serait mieux si EngineCommand ne connaisse pas AICommand et vice-versa. Cela nous évite une dépendance entre deux sous-modules de  deux modules séparés.

**Pull request merged**